### DISCUSSION TITLE
How does DIA-NN make the protein groups and can I make my own protein groups from unique peptides only using the pr.matrix file?

### DISCUSSION THREAD

**Discussion by DanielOliverDodd** [vdemichev/DiaNN#1080] [Q&A]
How does DIA-NN make the protein groups and can I make my own protein groups from unique peptides only using the pr.matrix file? I have no real coding experience so apologies if this is a very stupid question. I just want to understand where the protein groups have come from as I think there are some errors in the data I have gotten.

**Comment by vdemichev** [comment]
Hi Daniel,

You can do this if you have an implementation of some suitable algorithm, but this is not trivial. This normally also would be a bad idea, as DIA-NN's protein grouping also comes with q-values estimated for each protein group, i.e. if you replace the groups with something else you will not have q-values based on which you can filter.

The protein groups in DIA-NN are calculated using a maximum parsimony-based algorithm, the basic idea of this is nicely explained in the IDPicker paper (although the details of how this is implemented are very different in DIA-NN).

If you see some inconsistency in the data, please post here, can take a look.

Best,
Vadim

**Reply by DanielOliverDodd** [reply]
Hi Vadim,

Thanks very much for answering so quickly. So the project is to look at proteomes of cells over differentiation, we have a time course day 1, 4, 7, 14, 21, we also have matched RNA sequencing for these time points. Essentially what we see are some proteins appearing as high intensity in the protein groups but we do not think this is the case as we know from RNA that these proteins are not that high at this point in these cells. When I investigate the peptide file to see which peptides are contributing I see many fewer peptides for the protein in day 1 than in day 4 or 7 but the intensity in the protein group file is much higher.

I have attached a screenshot below to demonstrate what I mean. As you can see for the protein here (DNAH5) it looks as if the peptides are not particularly intense and there are very few peptides/no peptides for some samples when I have selected unique peptides only (proteolytic 1) however, when I take into account non-unique it significantly alters the data and increases the peptide intensity. Furthermore the sum of the peptide intensity is sometime 10x less than the protein group intensity and sometimes the same or higher. I think this is having a negative effect on my data as you can see from the attached graph, DNAH5 seems much higher at day 1 when it should broadly speaking follow the pattern of the other DNAHs and this seems to be due mostly to a single shared peptide. What is perplexing is that the shared peptide that is confounding day 1 in my samples is shared with a protein that is not expressed in these cells.

Sorry I cannot go into more detail right now but I can elaborate more later if you'd like me to.

![image](https://github.com/vdemichev/DiaNN/assets/149174861/4cb74896-e83a-45d8-81a7-27b13c13fc1b)

![image](https://github.com/vdemichev/DiaNN/assets/149174861/558eac96-6f58-4cc8-aee2-d7e7d677a644)

![Proteomics mTECs time course w DNAH5](https://github.com/vdemichev/DiaNN/assets/149174861/717e66fc-7b27-4db6-9d2b-47bf663a01a8)

**Reply by vdemichev** [reply]
Hi Daniel, what you describe is a known issue of quantification in proteomics. The q-values reported control FDR but do not control quantification accuracy. So even if you filter by PEP, then 1% filter means there's 99% chance that at least one of the fragment ions corresponding to the protein is matched to the correct cognate signal. As you can see, that does not guarantee at all that the quantity is correct. Further, the MaxLFQ algorithm is more sensitive to this than less precise Top 1 method, used to obtain PG.Normalised. 

So what I would recommend, if you would like to get higher confidence (based on the main .parquet report):
- Filter PG.Q.Value <= 0.01 or, if not enough, PG.PEP <= 0.01
- Filter PG.MaxLFQ.Quality > 0.9 (or higher) and Empirical.Quality > 0.7 (or higher). I would fix PG.MaxLFQ.Quality to 0.9 and see what is the highest level of Empirical.Quality that still produces good ID numbers.
- Optionally (requires some simple processing in R): only include identifications of a protein with >= N peptides (N = 2 or 3 is reasonable)
So the above I guess will just remove the identification at day 1, i.e. it will just not be reported.

In terms of what is proteotypic. Are you looking at Genes.MaxLFQ.Unique? This will be the quantity you want to use if in a particular case you believe you cannot rely on non-proteotypic peptides.

A very useful technique when you need to verify the expression pattern of certain proteins with high confidence is to look at individual peptides. Works particularly well if you have many peptides detected per protein overall. One option is to just add a supplementary figure which plots the trend (precursor intensity vs day) for each of the precursors (like a plot with many small panels, each panel a precursor). This is useful not only to exclude LC-MS errors, but also to detect a situation when there's differential regulation of proteoforms. One can think of also other options. E.g. split the set of peptides into two groups (can, say, order by average log-intensity, odd peptides go into one group, even go into another), recalculate protein quantities using Top 1 or MaxLFQ with each group of peptides, plot and show that the trend protein level vs day number is the same for each group. This massively reduces the chance of what you are seeing being an LC-MS artifact.

**Reply by DanielOliverDodd** [reply]
Hi Vadim,

Thanks very much for the detailed answer, I appreciate these suggests and we will try them out. I think we are looking at Genes.MAXLFQ.Unique, I am just the end user so I did not run the samples myself but I wanted to refine these results a bit. I'll let you know what works for us!

Many thanks,

Dan
